homeassistant/components/airzone/sensor.py

```diff
@@ -0,0 +1,76 @@
+"""Support for the Airzone sensors."""
+from __future__ import annotations
+
+from aioairzone.const import AZD_NAME, AZD_SYSTEMS, AZD_ZONES
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import AirzoneUpdateCoordinator
+from .const import DOMAIN, MANUFACTURER, SENSOR_TYPES
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Airzone sensors from a config_entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = []
+    for system_id, system_data in coordinator.data[AZD_SYSTEMS].items():
+        for zone_id, zone_data in system_data[AZD_ZONES].items():
+            zone_name = zone_data[AZD_NAME]
+
+            device_info: DeviceInfo = {
+                "identifiers": {(DOMAIN, f"{system_id},{zone_id}")},
+                "manufacturer": MANUFACTURER,
+                "name": f"Airzone [{system_id}:{zone_id}] {zone_name}",
+            }
+
+            for description in SENSOR_TYPES:
+                if description.key in zone_data:
+                    sensors.append(
+                        AirzoneSensor(
+                            coordinator,
+                            description,
+                            device_info,
+                            system_id,
+                            zone_id,
+                            zone_name,
+                        )
+                    )
```

Comment:
Can you rework this so there is no triple for loop?

Reply:
Should I split this to several functions or should I remove the triple loop?
Would a double loop be ok?
Something like:
```
for zone in zones:
  For sensor in sensors:
```

Reply:
@tkdrob I simplified the triple loop into a double loop. I hope that's enough, but tell me if you want me to change something else.